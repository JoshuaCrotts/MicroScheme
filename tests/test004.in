;
; Simple expressions being called in terms of each other.
;

(define (sq x) (* x x))
(sq 5)                          ; 25
(* (sq 9) 3)                    ; 243
(define (add x y) (+ x y))
(define (add_weird x y z) (+ 2 x y z))
(add (sq 9) (sq 10))            ; 181
(add_weird (add 4 5) (sq 5) 5)  ; 2 + (9 + 25 + 5) = (2 + 39) = 41
