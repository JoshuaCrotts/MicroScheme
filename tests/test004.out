25
243
181
41
